src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Document.cs

```diff
@@ -0,0 +1,173 @@
+        }
+
+        /// <summary>
+        /// Converts the <see cref="JsonDocument"/> representing a single JSON value into a <paramref name="returnType"/>.
+        /// </summary>
+        /// <returns>A <paramref name="returnType"/> representation of the JSON value.</returns>
+        /// <param name="document">The <see cref="JsonDocument"/> to convert.</param>
+        /// <param name="returnType">The type of the object to convert to and return.</param>
+        /// <param name="context">A metadata provider for serializable types.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// <paramref name="document"/> is <see langword="null"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="returnType"/> is <see langword="null"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="context"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="JsonException">
+        /// The JSON is invalid.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="returnType" /> is not compatible with the JSON.
+        ///
+        /// -or-
+        ///
+        /// There is remaining data in the string beyond a single JSON value.</exception>
+        /// <exception cref="NotSupportedException">
+        /// There is no compatible <see cref="System.Text.Json.Serialization.JsonConverter"/>
+        /// for <paramref name="returnType"/> or its serializable members.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// The <see cref="JsonSerializerContext.GetTypeInfo(Type)"/> method of the provided
+        /// <paramref name="context"/> returns <see langword="null"/> for the type to convert.
+        /// </exception>
+        public static object? Deserialize(this JsonDocument document, Type returnType, JsonSerializerContext context)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (returnType == null)
+            {
+                throw new ArgumentNullException(nameof(returnType));
+            }
+
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            return ReadUsingMetadata<object?>(document, GetTypeInfo(context, returnType));
+        }
+
+        [RequiresUnreferencedCode(SerializationUnreferencedCodeMessage)]
+        private static TValue? ReadUsingOptions<TValue>(JsonDocument document, Type returnType, JsonSerializerOptions? options) =>
+            ReadUsingMetadata<TValue>(document, GetTypeInfo(returnType, options));
+
+        private static TValue? ReadUsingMetadata<TValue>(JsonDocument document, JsonTypeInfo jsonTypeInfo)
+        {
+            ReadOnlySpan<byte> utf8Json = document.GetRawValue(0, includeQuotes: true).Span;
```

Comment:
Could this implementation have called document.RootElement and passed that to `ReadUsingMetadata<TValue>(JsonElement element, JsonTypeInfo jsonTypeInfo)`?

Reply:
Sure. Thanks